### PR TITLE
PIMS-1284 Adding Parcel PUT unit tests

### PR DESCRIPTION
**Pull request by Fosol**
Fixing bugs related to updating parcels.
Found various issues with Automapper behaviour.
Found various issues with EF CurrentValues.SetValues() behaviour.

Please review the other PR before this one.

### CODE REVIEW

File: backend/api/Areas/Admin/Controllers/ParcelController.cs [hunk 1]
```diff
@@ -152,7 +157,7 @@ public IActionResult AddParcel([FromBody] Model.ParcelModel model)
             _pimsAdminService.Parcel.Add(entity);
             var parcel = _mapper.Map<Model.ParcelModel>(entity);
 
-            return new CreatedAtActionResult(nameof(GetParcel), nameof(ParcelController), new { id = parcel.Id }, parcel);
+            return CreatedAtAction(nameof(GetParcel), new { id = parcel.Id }, parcel);
```
Comment (Fosol) [inline backend/api/Areas/Admin/Controllers/ParcelController.cs]: Fixing issue where it couldn't find the correct action...

File: backend/api/Profiles/BuildingProfile.cs [hunk 2]
```diff
@@ -36,10 +36,13 @@ public BuildingProfile()
...
                 .ForMember(dest => dest.AddressId, opt => opt.MapFrom(src => src.Address.Id))
+                .ForMember(dest => dest.Address, opt => opt.MapFrom(src => src.Address))
+                .ForMember(dest => dest.ParcelId, opt => opt.MapFrom(src => src.ParcelId))
                 .ForMember(dest => dest.Parcel, opt => opt.Ignore())
+                .ForMember(dest => dest.AgencyId, opt => opt.MapFrom(src => src.AgencyId))
                 .ForMember(dest => dest.Agency, opt => opt.Ignore())
-                .ForMember(dest => dest.IsSensitive, opt => opt.Ignore())
-                .ForMember(dest => dest.Evaluations, opt => opt.MapFrom(src => src.Evaluations))
+                .ForMember(dest => dest.IsSensitive, opt => opt.MapFrom(src => src.IsSensitive))
+                .ForMember(dest => dest.Evaluations, opt => opt.Ignore())
```
Comment (devinleighsmith) [inline backend/api/Profiles/BuildingProfile.cs]: I think this'll prevent my parcel update service call from saving building evaluations into the database.

Comment (Fosol) [reply backend/api/Profiles/BuildingProfile.cs]: It does on purpose.  Automapper will delete all buildings first then add new ones.  Not the desired behaviour.

Comment (Fosol) [reply backend/api/Profiles/BuildingProfile.cs]: I have updated the service layer so that it adds/updates/ buildings and their evaluations.

**Comment by asanchezr** [comment]
@Fosol could you please leave a comment why you closed the last two PRs here? Thanks!

**Comment by Fosol** [comment]
Changes required to fix Automapper and Concurrency require a new PR.

**Pull request closed without merging**